docs/_docs/09-ug-index-types.md

```diff
@@ -0,0 +1,315 @@
+
+### Index data layout
+
+The index data consists of indexed columns and included columns, and it is stored as bucketed and
+sorted within each bucket by indexed columns. "Bucketed" stands for "hash partitioned" simply;
+all rows with the same hash value go into the same bucket.
+We use 200 buckets by default which is the default value of `spark.sql.shuffle.partitions` config
+to increase the possibility of index application. 
+The bucket number for index data is configurable via `spark.hyperspace.index.numBuckets`.
+
+For example, assume the source data is stored in 1000 files in random order.
+At index creation time, Hyperspace performs shuffling and sorting by the indexed columns before
+storing index data.
+As a result, the index data consists of 200 files (bucket number) by default and within each bucket,
+rows are sorted by the indexed columns.
+
+### Applicable queries
+
+#### Sort Merge Join
+
+Sort merge join is a well known expensive operation in Spark because it requires full shuffling
+of both left and right dataset. We can optimize the time for shuffling and sorting datasets, as we
+already did the required data rearrangement at index creation time; by reading the index data
+instead of source data, Spark will skip shuffling & sorting for the join.
+Therefore, if some dataset is repeatedly used for Sort merge joins causing shuffles for every query
+time, creating a Covering Index would be helpful.
+We also support [mutable dataset](https://microsoft.github.io/hyperspace/docs/ug-mutable-dataset/)
+for covering index types.
+
+#### Filter conditions for the first indexed column
+
+Queries including filter conditions for the first indexed column can also be accelerated, since
+the index data is also sorted within each bucket. Thanks to statistics and data skipping features
+of Parquet format, the queries can be accelerated by skipping unnecessary row groups while reading
+the data. It could be less effective compared to global sorting, and only applicable for the first
+indexed column.
+
+### Index creation API
+
+#### CoveringIndexConfig
+
+To create a Covering Index, `CoveringIndexConfig` should be used at index creation with:
+- index name: name of index; should be unique within an index system directory.
+- indexed columns: columns to use for shuffling and sorting.
+- included columns: columns to include in the index data.
+
+#### Scala
+```scala
+import com.microsoft.hyperspace._
+import com.microsoft.hyperspace.index.covering._
+
+val df = spark.read.parquet("<path/to/data>")
+val idxConfig = CoveringIndexConfig(
+  "<indexName>",
+  Seq("<indexedCol1>", "<indexedCol2>"),
+  Seq("<includedCol1>"))
+val hs = new Hyperspace(sparkSession)
+hs.createIndex(df, idxConfig)
+```
+
+#### Python
+```python
+from hyperspace import Hyperspace
+from hyperspace import CoveringIndexConfig
+hs = Hyperspace(sparkSession)
+idxConfig = CoveringIndexConfig('<indexName>', ['<indexedCol1>', '<indexedCol2>'], ['<includedCol1>'])
```

Comment:
Using double quote will be more consistent with the following line. (And I personally prefer double quote 😄)
`'<indexName>'` -> `"<indexName>"`